### PR TITLE
Deduplicate alert manager alerts by caching their hashes

### PR DESCRIPTION
**Pull request by RobertSzefler**

### CODE REVIEW

File: src/robusta/runner/web.py [hunk 1]
```diff
@@ -57,11 +69,40 @@ def handle_alert_event():
         alert_manager_event = AlertManagerEvent(**req_json)
         for alert in alert_manager_event.alerts:
             alert = Web._relabel_alert(alert)
-            Web.alerts_queue.add_task(Web.event_handler.handle_trigger, PrometheusTriggerEvent(alert=alert))
+            alert_hash = Web.get_compound_hash([
+                alert.fingerprint.encode('ascii'),
+                alert.status.encode('utf-8'),
+                str(alert.startsAt.timestamp()).encode('ascii'),
```
Comment (arikalon1) [inline src/robusta/runner/web.py:75]: please also add the `endsAt`
I don't think you can get 2 alerts with everything the same, and only endsAt is different, but I don't see a downside in adding it

File: src/robusta/runner/web.py [hunk 2]
```diff
@@ -57,11 +69,40 @@ def handle_alert_event():
...
+                    Web.processed_alerts_cache[alert_hash] = True
+            Web.alerts_queue.add_task(
+                Web.event_handler.handle_trigger, PrometheusTriggerEvent(alert=alert)
+            )
 
         Web.event_handler.get_telemetry().last_alert_at = str(datetime.now())
         return jsonify(success=True)
 
+    @staticmethod
+    def processed_alerts_cache_flusher():
```
Comment (arikalon1) [inline src/robusta/runner/web.py:90]: I think we don't need this.
I think the TTL Cache handles expiration

File: src/robusta/runner/web.py [hunk 3]
```diff
@@ -26,13 +32,19 @@ class Web:
...
+    processed_alerts_cache_lock = threading.Lock()
+
     @staticmethod
     def init(event_handler: PlaybooksEventHandler, loader: ConfigLoader):
         Web.metrics = QueueMetrics()
         Web.api_server_queue = TaskQueue(name="api_server_queue", num_workers=NUM_EVENT_THREADS, metrics=Web.metrics)
         Web.alerts_queue = TaskQueue(name="alerts_queue", num_workers=NUM_EVENT_THREADS, metrics=Web.metrics)
         Web.event_handler = event_handler
         Web.loader = loader
+        threading.Thread(
```
Comment (arikalon1) [inline src/robusta/runner/web.py:45]: I think we don't need this

**Review by arikalon1 — COMMENTED** [review]
nice work
left few comments

File: src/robusta/runner/web.py (hunk 1 above)
Comment (RobertSzefler) [reply src/robusta/runner/web.py:75]: Added

File: src/robusta/runner/web.py (hunk 2 above)
Comment (RobertSzefler) [reply src/robusta/runner/web.py:91]: As discussed, this is implemented to handle some worst case scenarios when the built-in expiration mechanism in `TTLCache` would be run and take a very long time. To guard against this, we run `expire` periodically to distribute the workload.

Comment (RobertSzefler) [reply src/robusta/runner/web.py]: Okay, upon further inspection it looks like the "bulk expire", if it happens, in practice won't take so long to warrant having this mechanism implemented. Removed.

**Pull request merged**